ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py

```diff
@@ -108,3 +108,54 @@ def test_numpy_tanh(
         subok=True,
         test_values=False,
     )
+
+
+# arcsinh
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=ivy_np.valid_float_dtypes),
+    dtype=st.sampled_from(ivy_np.valid_float_dtypes + (None,)),
+    where=np_frontend_helpers.where(),
```

Comment:
This way of generating where will cause the error of where having different size with the x array, so you should use a function like this:
@st.composite
def _dtype_x_axis(draw, **kwargs):
    dtype, x, shape = draw(helpers.dtype_and_values(**kwargs, ret_shape=True))
    axis = draw(st.one_of(helpers.ints(min_value=0, max_value=max(len(shape) - 1, 0)), 
                          st.none()))
    where = draw(st.one_of(helpers.array_values(dtype=ivy.bool, shape=shape), 
                           st.none()))
    return (dtype, x, axis), where
to generate those arguments.

Reply:
Isn't this function similar to the fronted Numpy tan function as implemented on this link: https://lets-unify.ai/ivy/deep_dive/17_ivy_frontends_tests.html ?